main_test.go

```diff
@@ -0,0 +1,303 @@
+package main
+
+import (
+  "fmt"
+  _ "github.com/jinzhu/gorm/dialects/sqlite"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+func setUp() *App {
+
+  app := &App{}
+
+  app.Initialize("sqlite3", ":memory:")
+
+  return app
+}
+
+func tearDown(app *App) {
+  app.DB.Close()
+}
+
+func TestAddSong(t *testing.T) {
+  app := setUp()
+
+  testBody :=  "{\"Title\": \"blue\", \"SpotifyId\": \"12345\", \"URL\": \"www.moose.com\", \"Delay\": 2, \"AvBarDuration\": 3, \"Duration\": 123, \"Tempo\": 4, \"TimeSignature\": 8}"
+
+  stringifiedBody := strings.NewReader(testBody)
+
+  //if I hadn't written this as a string already, I'd have had to use the .Encode method
+  router := app.MakeRouter()
+
+  req, err := http.NewRequest("POST", "/api/v1/songs/", stringifiedBody)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Our API expects a form body, so set the content-type header to make sure it's treated as one.
+	req.Header.Add("Content-Type", "application/json")
+
+	res := httptest.NewRecorder()
+
+	router.ServeHTTP(res, req)
+
+  // assert.Equal(t, http.StatusOK, rr.Code)
```

Comment:
I just wanted to add this in case I ever want to try this with an assertion framework!